### DISCUSSION TITLE
After counter(page).update(), query() returns elements whose location().page() returns the original (physical) page number

### DISCUSSION THREAD

**Discussion by DrGo** [typst/typst#4198] [General]
### Description

After resetting the page counter, e.g.,  ` counter(page).update(1)`,

`counter(page).get().first()` returns the expected page number  (the logical number based on location in the document where the counter was updated).

query(), however, continues to return elements with the original (physical) page number as if the page counter was not updated. e.g., after
``` typst
let elems = query(selector(heading).before(here()),)
```
`elems[x].location.page()` returns the physical page number.

So, it is not possible to compare its results with the result of ` counter(page).get().first()`, e.g., to output different page headers based on *both* the current page number AND the contents of previous headings.

Am I missing something here? Is there an alternative way to achieve what I want? 

### Reproduction URL

https://typst.app/project/rCPRoO_AdU7tV0CH1eaeLx

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
This is intentional. The page number is the physical page number. You'll have to to use `counter(page).at(location)` if you want the counter.

See `page` under https://typst.app/docs/reference/introspection/location/
